### PR TITLE
Feature/m2-725 item name and item body

### PR DESCRIPTION
**Pull request by alex-i-e**
The uploading integration will be added in a separate PR

### CODE REVIEW

File: src/svgSprite.ts [hunk 1]
```diff
@@ -907,5 +907,31 @@ export const svgSprite = (): string => `
     <rect width="72" height="4" rx="2" fill="#DEE3EB"/>
     <rect width="41" height="4" rx="2" fill="#42474E"/>
   </symbol>
+  <symbol viewBox="0 0 12 16" id="svg-md-editor-mark">
```
Comment (nmialik) [inline src/svgSprite.ts:910]: Can you add empty lines between symbols, please?
And please import icons 24x24, you can find them [here](https://www.figma.com/file/LXl9BAJSwtgWuzkxqOqvX8/ML1-Design-System?node-id=53015-36793&t=AR0FKTZEKXAlglih-0)

File: src/components/MarkDownEditor/extensions/VideoUploadExtension/VideoUploadExtension.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import React, { FC } from 'react';
+import MdEditor, { InsertContentGenerator } from 'md-editor-rt';
+import { Paper } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+
+import { Svg } from 'components/Svg';
```
Comment (nmialik) [inline src/components/MarkDownEditor/extensions/VideoUploadExtension/VideoUploadExtension.tsx:6]: please use just `import { Svg } from 'components';`

File: src/components/MarkDownEditor/extensions/ImageUploadExtension/ImageUploadExtension.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import React, { FC } from 'react';
+import MdEditor, { InsertContentGenerator } from 'md-editor-rt';
+import { Paper } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+
+import { Svg } from 'components/Svg';
```
Comment (nmialik) [inline src/components/MarkDownEditor/extensions/ImageUploadExtension/ImageUploadExtension.tsx:6]: `import { Svg } from 'components';`

File: src/components/MarkDownEditor/extensions/extensions.styles.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  padding: 0;
+  border-radius: 3px;
+  border: 1px solid var(--md-border-color);
+  background-color: inherit;
+`;
+
+export const StyledMenuItem = styled(MenuItem)`
+  font-size: 12px;
+  color: var(--md-color);
+  padding: 4px 10px;
```
Comment (nmialik) [inline src/components/MarkDownEditor/extensions/extensions.styles.ts:15]: I suppose we can use variables and theme here

font-size: ${variables.font.size.sm};
padding: ${theme.spacing(0.4, 1)};

File: src/components/MarkDownEditor/SourceLinkModal/SourceLinkModal.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import React, { FC } from 'react';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+import { Modal } from 'components/index';
```
Comment (nmialik) [inline src/components/MarkDownEditor/SourceLinkModal/SourceLinkModal.tsx:5]: `import { Modal } from 'components';`

File: src/components/MarkDownEditor/extensions/AlignTextExtension/AlignTextExtension.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import React, { FC } from 'react';
+import MdEditor, { InsertContentGenerator } from 'md-editor-rt';
+
+import { Svg } from 'components/Svg';
```
Comment (nmialik) [inline src/components/MarkDownEditor/extensions/AlignTextExtension/AlignTextExtension.tsx:4]: import { Svg } from 'components';

File: src/components/MarkDownEditor/extensions/AudioUploadExtension/AudioUploadExtension.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
+import React, { FC } from 'react';
+import MdEditor, { InsertContentGenerator } from 'md-editor-rt';
+import { Paper } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+
+import { Svg } from 'components/Svg';
```
Comment (nmialik) [inline src/components/MarkDownEditor/extensions/AudioUploadExtension/AudioUploadExtension.tsx:6]: `import { Svg } from 'components';`

File: src/components/FormComponents/EditorController/EditorController.tsx [hunk 8]
```diff
@@ -1,28 +1,115 @@
...
+import 'md-editor-rt/lib/style.css';
 
-import { StyledEditor } from './EditorController.styles';
+import { MarkExtension } from 'components/MarkDownEditor/extensions/MarkExtension';
+import { TrashExtension } from 'components/MarkDownEditor/extensions/TrashExtension';
+import { AlignTextExtension } from 'components/MarkDownEditor/extensions/AlignTextExtension';
+import { AudioUploadExtension } from 'components/MarkDownEditor/extensions/AudioUploadExtension';
+import { ImageUploadExtension } from 'components/MarkDownEditor/extensions/ImageUploadExtension';
+import { VideoUploadExtension } from 'components/MarkDownEditor/extensions/VideoUploadExtension';
+import { LANGUAGE_BY_DEFAULT } from 'components/MarkDownEditor';
```
Comment (nmialik) [inline src/components/FormComponents/EditorController/EditorController.tsx:13]: Maybe we can add index file to import all extensions from one place?

File: src/components/MarkDownEditor/extensions/VideoUploadExtension/VideoUploadExtension.tsx (hunk 2 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/extensions/VideoUploadExtension/VideoUploadExtension.tsx]: done

File: src/components/MarkDownEditor/extensions/ImageUploadExtension/ImageUploadExtension.tsx (hunk 3 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/extensions/ImageUploadExtension/ImageUploadExtension.tsx]: done

File: src/components/MarkDownEditor/extensions/extensions.styles.ts (hunk 4 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/extensions/extensions.styles.ts]: done

File: src/components/MarkDownEditor/SourceLinkModal/SourceLinkModal.tsx (hunk 5 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/SourceLinkModal/SourceLinkModal.tsx]: done

File: src/components/MarkDownEditor/extensions/AlignTextExtension/AlignTextExtension.tsx (hunk 6 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/extensions/AlignTextExtension/AlignTextExtension.tsx]: done

File: src/components/MarkDownEditor/extensions/AudioUploadExtension/AudioUploadExtension.tsx (hunk 7 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/extensions/AudioUploadExtension/AudioUploadExtension.tsx]: done

File: src/components/FormComponents/EditorController/EditorController.tsx (hunk 8 above)
Comment (alex-i-e) [reply src/components/FormComponents/EditorController/EditorController.tsx]: done

File: src/components/FormComponents/EditorController/EditorController.styles.ts [hunk 9]
```diff
@@ -1,40 +1,49 @@
...
-export const StyledEditor = styled(MDEditor)`
-  &.w-md-editor {
-    border-radius: ${variables.borderRadius.lg2};
-    border: ${variables.borderWidth.md} solid ${variables.palette.surface_variant};
-    background-color: ${variables.palette.surface};
-    color: ${variables.palette.on_surface_variant};
-    font-size: ${variables.font.size.lg};
-    box-shadow: unset;
-  }
+import { MdEditor } from 'components/MarkDownEditor';
```
Comment (IhorNesterenko99) [inline src/components/FormComponents/EditorController/EditorController.styles.ts:6]: pls add to export to components/index to import MdEditor from components
and remove empty space between styles and components import

File: src/components/FormComponents/EditorController/EditorController.tsx [hunk 10]
```diff
@@ -1,28 +1,117 @@
...
+            '-',
+            'catalog',
+            'preview',
+            'pageFullscreen',
+            'htmlPreview',
+          ]}
+          footers={[0, '=', 1]}
+          defFooters={[
+            <FooterMessage inputSize={value.toString().length} key="footer-message" />,
+            // '=',
```
Comment (IhorNesterenko99) [inline src/components/FormComponents/EditorController/EditorController.tsx:110]: Maybe this commented code is not needed?

File: src/components/MarkDownEditor/CharacterCounter/CharacterCounter.tsx [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import React, { FC } from 'react';
+
+import { StyledTitleSmall } from 'styles/styledComponents';
+import theme from 'styles/theme';
+
+import { THRESHOLD_SIZE } from '../MarkDownEditor';
+import { CharacterCounterProps } from './CharacterCounter.types';
+
+const CharacterCounter: FC<CharacterCounterProps> = ({ inputSize }) => {
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/CharacterCounter/CharacterCounter.tsx:9]: pls use {}: CharacterCounterProps (without FC)

File: src/components/MarkDownEditor/FooterMessage/FooterMessage.styles.ts [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import { styled } from '@mui/system';
+
+import { StyledTitleSmall } from 'styles/styledComponents';
+import { variables } from 'styles/variables';
+import theme from 'styles/theme';
+
+export const StyledText = styled(StyledTitleSmall)`
+  color: ${variables.palette.red};
+  margin: 0 ${theme.spacing(1)};
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/FooterMessage/FooterMessage.styles.ts:9]: pls use theme.spacing(0, 1)

File: src/components/MarkDownEditor/FooterMessage/FooterMessage.tsx [hunk 13]
```diff
@@ -0,0 +1,22 @@
+import React, { FC } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { FooterMessageProps } from './FooterMessage.types';
+import { StyledText } from './FooterMessage.styles';
+import { THRESHOLD_SIZE } from '../MarkDownEditor';
+
+const FooterMessage: FC<FooterMessageProps> = ({ inputSize }) => {
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/FooterMessage/FooterMessage.tsx:8]: without FC

File: src/components/MarkDownEditor/extensions/extensions.hooks.ts [hunk 14]
```diff
@@ -0,0 +1,44 @@
+import { useRef, useState } from 'react';
```
Comment (kdanilenka32) [inline src/components/MarkDownEditor/extensions/extensions.hooks.ts:1]: space

File: src/components/MarkDownEditor/MarkDownEditor.ts [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+
+export const LANGUAGE_BY_DEFAULT = 'en';
+export const THRESHOLD_SIZE = 75;
+
+MdEditor.config({
+  editorConfig: {
+    languageUserDefined: {
+      [LANGUAGE_BY_DEFAULT]: {
+        toolbarTips: {
+          bold: i18n.t('mdEditorBold'),
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/MarkDownEditor.ts:12]: use const { t } = i18n and then use just t('...')

File: src/components/MarkDownEditor/MarkDownEditor.ts [hunk 16]
```diff
@@ -0,0 +1,47 @@
+import MdEditor from 'md-editor-rt';
+import i18n from 'i18n';
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/MarkDownEditor.ts:2]: pls add spacing between npm and components import

File: src/components/MarkDownEditor/SourceLinkModal/SourceLinkModal.tsx [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+import { InputController } from 'components/FormComponents';
+
+import {
+  SourceLinkModalForm,
+  SourceLinkModalFormValues,
+  SourceLinkModalProps,
+} from './SourceLinkModal.types';
+import { StyledController, StyledModalWrapper } from './SourceLinkModal.styles';
+
+const SourceLinkModal: FC<SourceLinkModalProps> = ({ title, handleClose, handleSubmit }) => {
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/SourceLinkModal/SourceLinkModal.tsx:15]: without FC and add export

File: src/components/MarkDownEditor/extensions/AlignTextExtension/AlignTextExtension.tsx [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+
+import { Svg } from 'components';
+import { StyledFlexAllCenter } from 'styles/styledComponents';
+import theme from 'styles/theme';
+
+import { AlignTextExtensionProps } from './AlignTextExtension.types';
+
+const NormalToolbar = MdEditor.NormalToolbar;
+
+const AlignTextExtension: FC<AlignTextExtensionProps> = ({
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/extensions/AlignTextExtension/AlignTextExtension.tsx:12]: export const and without FC

File: src/components/MarkDownEditor/extensions/ImageUploadExtension/ImageUploadExtension.tsx [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+import theme from 'styles/theme';
+
+import { StyledMenuItem, StyledMenuList } from '../extensions.styles';
+import { SourceLinkModal, SourceLinkModalForm } from '../../SourceLinkModal';
+import { useUploadMethods } from '../extensions.hooks';
+import { InsertContentExtensionProps } from '../extensions.types';
+
+const DropdownToolbar = MdEditor.DropdownToolbar;
+
+const ImageUploadExtension: FC<InsertContentExtensionProps> = ({ onInsert }) => {
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/extensions/ImageUploadExtension/ImageUploadExtension.tsx:17]: the same

File: src/components/MarkDownEditor/extensions/ImageUploadExtension/ImageUploadExtension.tsx [hunk 20]
```diff
@@ -0,0 +1,88 @@
+import React, { FC } from 'react';
+import MdEditor, { InsertContentGenerator } from 'md-editor-rt';
+import { Paper } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+
+import { Svg } from 'components';
+import { StyledFlexAllCenter, StyledFlexColumn, StyledTitleSmall } from 'styles/styledComponents';
+import theme from 'styles/theme';
+
+import { StyledMenuItem, StyledMenuList } from '../extensions.styles';
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/extensions/ImageUploadExtension/ImageUploadExtension.tsx:10]: Why not to use Extensions with upper cased first letter?

File: src/components/MarkDownEditor/extensions/MarkExtension/MarkExtension.tsx [hunk 21]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/extensions/MarkExtension/MarkExtension.tsx:1]: there is no need to import React itself, also in other files

File: src/components/MarkDownEditor/extensions/MarkExtension/MarkExtension.tsx [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <NormalToolbar
+      title={t('mdEditorMark')}
+      onClick={markHandler}
+      trigger={
+        <StyledFlexAllCenter sx={{ p: theme.spacing(0, 0.4) }}>
+          <Svg id="md-editor-mark" width="16" height="16" />
+        </StyledFlexAllCenter>
+      }
+    ></NormalToolbar>
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/extensions/MarkExtension/MarkExtension.tsx:35]: pls change to <NormalToolbar ... />

File: src/components/MarkDownEditor/extensions/MarkExtension/MarkExtension.tsx [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+
+import { Svg } from 'components';
+import { StyledFlexAllCenter } from 'styles/styledComponents';
+import theme from 'styles/theme';
+
+import { InsertContentExtensionProps } from '../extensions.types';
+
+const NormalToolbar = MdEditor.NormalToolbar;
+
+const MarkExtension = ({ onInsert }: InsertContentExtensionProps) => {
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/extensions/MarkExtension/MarkExtension.tsx:13]: pls add export const ... here

File: src/components/MarkDownEditor/extensions/TrashExtension/TrashExtension.tsx [hunk 24]
```diff
@@ -0,0 +1,29 @@
...
+
+import { Svg } from 'components';
+import { StyledFlexAllCenter } from 'styles/styledComponents';
+import theme from 'styles/theme';
+
+import { TrashExtensionProps } from './TrashExtension.types';
+
+const NormalToolbar = MdEditor.NormalToolbar;
+
+const TrashExtension: FC<TrashExtensionProps> = ({ onClick }) => {
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/extensions/TrashExtension/TrashExtension.tsx:13]: the same

File: src/components/FormComponents/EditorController/EditorController.tsx [hunk 25]
```diff
@@ -1,28 +1,117 @@
...
+  LANGUAGE_BY_DEFAULT,
+  AlignTextExtension,
+  AudioUploadExtension,
+  CharacterCounter,
+  ImageUploadExtension,
+  FooterMessage,
+  TrashExtension,
+  VideoUploadExtension,
+  MarkExtension,
+} from 'components/MarkDownEditor';
```
[the remark this answers is not in the document]
Comment (alex-i-e) [reply src/components/FormComponents/EditorController/EditorController.tsx:17]: done

File: src/components/MarkDownEditor/extensions/TrashExtension/TrashExtension.tsx [hunk 26]
```diff
@@ -0,0 +1,29 @@
+import React, { FC } from 'react';
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/extensions/TrashExtension/TrashExtension.tsx:1]: no need in these imports, also in other files

File: src/components/MarkDownEditor/FooterMessage/FooterMessage.styles.ts (hunk 12 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/FooterMessage/FooterMessage.styles.ts:9]: done

File: src/components/MarkDownEditor/extensions/VideoUploadExtension/VideoUploadExtension.tsx [hunk 27]
```diff
@@ -0,0 +1,88 @@
...
+import theme from 'styles/theme';
+
+import { StyledMenuItem, StyledMenuList } from '../extensions.styles';
+import { SourceLinkModal, SourceLinkModalForm } from '../../SourceLinkModal';
+import { useUploadMethods } from '../extensions.hooks';
+import { InsertContentExtensionProps } from '../extensions.types';
+
+const DropdownToolbar = MdEditor.DropdownToolbar;
+
+const VideoUploadExtension: FC<InsertContentExtensionProps> = ({ onInsert }) => {
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/extensions/VideoUploadExtension/VideoUploadExtension.tsx:17]: the same

File: src/components/MarkDownEditor/extensions/extensions.hooks.ts [hunk 28]
```diff
@@ -0,0 +1,44 @@
...
+  const onAddLinkClick = () => {
+    setIsPopupVisible(true);
+  };
+
+  const onUploadClick = () => {
+    inputRef.current?.click();
+  };
+
+  const onInputChange = () => {
+    // @todo: add uploading logic
```
Comment (IhorNesterenko99) [inline src/components/MarkDownEditor/extensions/extensions.hooks.ts:30]: pls use // TODO: ...

File: src/components/MarkDownEditor/FooterMessage/FooterMessage.tsx (hunk 13 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/FooterMessage/FooterMessage.tsx:8]: done

File: src/components/MarkDownEditor/extensions/extensions.hooks.ts (hunk 14 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/extensions/extensions.hooks.ts:1]: done

File: src/resources/app-en.json [hunk 29]
```diff
@@ -511,5 +513,50 @@
   "textInputOptionDescription": "The respondent will be able to enter a text response",
   "textInputOptionRequired": "Required",
   "minValue": "Min value",
-  "maxValue": "Max value"
+  "maxValue": "Max value",
+  "mdEditorBold": "Bold",
```
Comment (IhorNesterenko99) [inline src/resources/app-en.json:517]: I think it's better to use the key without mdEditor prefix, there could be cases that have the same value, for example just bold

File: src/components/MarkDownEditor/MarkDownEditor.ts (hunk 16 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/MarkDownEditor.ts:2]: this is correct, they are both from npm.

File: src/components/MarkDownEditor/SourceLinkModal/SourceLinkModal.tsx (hunk 17 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/SourceLinkModal/SourceLinkModal.tsx:15]: done

File: src/components/MarkDownEditor/extensions/AlignTextExtension/AlignTextExtension.tsx (hunk 18 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/extensions/AlignTextExtension/AlignTextExtension.tsx:12]: done

File: src/components/MarkDownEditor/extensions/ImageUploadExtension/ImageUploadExtension.tsx (hunk 19 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/extensions/ImageUploadExtension/ImageUploadExtension.tsx:17]: done

File: src/components/MarkDownEditor/extensions/ImageUploadExtension/ImageUploadExtension.tsx (hunk 20 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/extensions/ImageUploadExtension/ImageUploadExtension.tsx:10]: done

File: src/components/MarkDownEditor/extensions/MarkExtension/MarkExtension.tsx (hunk 22 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/extensions/MarkExtension/MarkExtension.tsx:35]: done

File: src/components/MarkDownEditor/extensions/MarkExtension/MarkExtension.tsx (hunk 23 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/extensions/MarkExtension/MarkExtension.tsx:13]: done

File: src/components/MarkDownEditor/extensions/TrashExtension/TrashExtension.tsx (hunk 24 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/extensions/TrashExtension/TrashExtension.tsx:13]: done

File: src/components/MarkDownEditor/extensions/VideoUploadExtension/VideoUploadExtension.tsx (hunk 27 above)
Comment (alex-i-e) [reply src/components/MarkDownEditor/extensions/VideoUploadExtension/VideoUploadExtension.tsx:17]: done

File: src/svgSprite.ts (hunk 1 above)
Comment (alex-i-e) [reply src/svgSprite.ts:911]: done

File: src/resources/app-en.json (hunk 29 above)
Comment (alex-i-e) [reply src/resources/app-en.json]: done

**Review by kdanilenka32 — APPROVED** [review]

File: src/components/MarkDownEditor/MarkDownEditor.ts (hunk 16 above)
Comment (IhorNesterenko99) [reply src/components/MarkDownEditor/MarkDownEditor.ts:2]: it's not, because we have i18n.ts file in src

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**